utils/credentialsUtils.js

```diff
@@ -0,0 +1,145 @@
+
+async function encryptAndPrepareMessage(senderDID, receiverDID, credentialsSubject) {
+
+  console.log("encryptAndPrepareMessage");
+
+  const encryptionPayload = {
+    senderDid: senderDID,
+    recipientDids: [receiverDID],
+    type: "issue",
+    payload: {
+      domain: "api.dock.io",
+      credentials: credentialsSubject
+    }
+  };
+
+  console.log("encryptionPayload:", encryptionPayload);
+
+  const data = await postRequest(`${dockUrl}/messaging/encrypt`, encryptionPayload)
+
+  console.log("didComm Message:", data);
+
+  return data.data.jwe;
+}
+
+const signedCredential = async (receiverDid) => {
+  const enrollmentId = `${uuidv4()}`;;
+
+  const bankIdData = {
+    receiverDid: receiverDid,
+    receiverName: "John Smith", //Name input
+    receiverAddress: "123 Main St, Anytown, CA 90210",
+    enrollmentId: enrollmentId
+  };
+
+  const biometricsData = {
+    receiverDid: receiverDid,
+    enrollmentId: enrollmentId,
+    biometricData: "some_biometrics_data",
+    enrollmentDate: "2024-02-05",
+    validTo: "2024-03-05",
+  };
+
+  const creditScoreData = {
+    receiverDid: receiverDid,
+  };
+
+  const bankIdCredential = createBankIdCredential(bankIdData);
+  const creditScoreCredential = createCreditScoreCredential(creditScoreData)
+  const biometricsCredential = createBiometricsCredential(biometricsData);
+
+  const credentials = [
+    bankIdCredential,
+    creditScoreCredential,
+    biometricsCredential
+  ];
+
+  const issuedCredentials = await Promise.all(credentials.map(async credential => {
+    const credentialData = await postRequest(credential.url, credential.body)
+    return credentialData.data
+  }));
+
+  return issuedCredentials;
+};
+
+export const issueCredentials = async (
+  receiverDID,
+  setIsLoading,
+  setIsSuccess
+) => {
+
+  const responses = [];
+
+  try {
+    const issuedCredentials = await signedCredential(receiverDID);
+
+    const senderDIDMapping = [
+      "did:dock:5HKkVpaciu1RArV13E7ig3i84JtiMTcwoXoHPZ8VMrBUYJ4w",
+      "did:dock:5HLbQLSmirNuZVRsdWKbsgdajw9QTGzSFJABSVzMT5EBj5sb",
+      "did:dock:5CKsfvaE68mvRhdn3dDXG4KpWzuvaUNdBbiu6sFUuPK9rw66"
+    ];
+
+    console.log("issuedCredentials: ", issuedCredentials);
+
+    for (let i = 0; i < issuedCredentials.length; i++) {
+      const issuedCredential = issuedCredentials[i];
+
+      // Get the senderDID from the mapping using the index
+      const senderDID = senderDIDMapping[i];
+
+      console.log("Processing credential:", issuedCredential);
+
+      console.log("Processing credential senderDID:", senderDID);
+
+      const encryptedMessage = await encryptAndPrepareMessage(senderDID, receiverDID, [issuedCredential.credentialSubject]);
+
+      console.log("encryptedMessage:", encryptedMessage);
+
+      const sendMessagePayload = {
+        to: receiverDID,
+        message: encryptedMessage
+      };
+
+      const data = await postRequest(`${dockUrl}/messaging/send`, sendMessagePayload)
```

Comment:
@cykoder they shouldn't have to do this manually, right? If there's a DID specified in the `credentialSubject.id` field we'll do it automatically for them

Reply:
yeah, also must provide distribute: true

Reply:
inside _credentials folder all the crendentials js, contain distribute "true"
![2024-02-12_14h11_36](https://github.com/docknetwork/dock-bank-demo/assets/23556116/1df1010c-c3c6-4930-9bf5-6f12ed8e9947)